onnxruntime/python/tools/quantization/test/test_calibrate.py

```diff
@@ -7,130 +7,349 @@
 # --------------------------------------------------------------------------
 
 import unittest
-import os
 import onnx
 import onnxruntime
-from onnx import helper, TensorProto
-from onnx.helper import make_node, make_tensor_value_info
-import calibrate
 import numpy as np
-from onnx import numpy_helper
+from onnx import helper, TensorProto, numpy_helper
+from onnxruntime.quantization.calibrate import calibrate, CalibrationDataReader, ONNXCalibrater
+
+
+def generate_input_initializer(tensor_shape,tensor_dtype,input_name):
+  '''
+  Helper function to generate initializers for test inputs
+  '''
+  tensor = np.random.ranf(tensor_shape).astype(tensor_dtype)
+  init = numpy_helper.from_array(tensor,input_name)
+  return init  
+
+class TestDataReader(CalibrationDataReader):
+    '''for test purpose'''
+    def __init__(self, calibration_image_folder):
+        self.image_folder = calibration_image_folder
+
+    def get_next(self):
+        return None
+
+class TestDataReaderSecond(CalibrationDataReader):
+    '''for test purpose'''
+    def __init__(self, calibration_image_folder):
+        self.image_folder = calibration_image_folder
+        self.preprocess_flag = True
+        self.enum_data_dicts = []
+        self.datasize = 3
+
+    def get_next(self):
+        if self.preprocess_flag:
+            self.preprocess_flag = False
+            nhwc_data_list = []
+            for i in range(3):
+                nhwc_data_list.append(np.random.ranf([1,3,1,3]).astype(np.float32))
```

Comment:
I would use a static data instead of random data, so that we can check the reduce_min and reduce_max in following step